### ISSUE TITLE
RTMP third party client API will be banned

### ISSUE THREAD

**Issue by Divi** [loldevs/leaguespec#30]
See : https://developer.riotgames.com/discussion/riot-games-api/show/oklxAP21

**TL;DR:** Riot will block all RTMP third party API. The only way to retrieve game data will be with their official API, which is very poor at this time. Retrieve the spectating encryption key will be impossible. Maybe they will release an API with this data, I don't know.

**Comment by jaagupkymmel** [comment]
http://www.reddit.com/r/leagueoflegends/comments/2easbn/riot_wants_any_3rd_party_applications_like/cjxpb3r

Apparently Riot will allow to continue downloading spectator data, so if they are using the same method that they are, we should be in the safe. Should confirm with a rioter tho.

**Comment by Divi** [comment]
You're right, but the encryption key is not in the meta, it's in the RTMP API. And we won't be able to know the live game id.

**Comment by lukegb** [comment]
From my reading: They're not blocking it, they're just going to start enforcing the don't-use-APIs-we-haven't-provided rule. They then went on (in edit 2) to explicitly spell out that _exclusively for spectating stuff_ you can retain this functionality and continue to use, for the time being, the RTMP API, since that's the only way to actually get the data you need to do that.

**Comment by avborhanian** [comment]
Yeah, I thought it was okay because of this comment. "I meant, they are welcome to keep using it as they have been, as a solution until some future point where we can potentially bring this functionality into the API. This is less of an exception for a particular use case, and more of the point that what they are doing does not affect our platform in the way the more common scraping methods do and is therefore not a concern at this time." Seems to imply RTMP is okay, but they will eventually move it to the Riot API.

**Comment by Divi** [comment]
I don't think so, they say OP.GG and lolking have exclusive API (private API), which has the spectator API.

**Comment by jaagupkymmel** [comment]
@Divi source? Should I message a rioter about this (possibly RiotSargonas)?

**Comment by Divi** [comment]
It was in the Reddit thread. Sargonas said big webites has moved on an "exclusive" API. I already send a message to Riot Sargonas, but no answer atm.

**Comment by frostycpu** [comment]
@Divi @jaagupkymmel this: https://www.reddit.com/r/leagueoflegends/comments/2easbn/riot_wants_any_3rd_party_applications_like/cjxp2d3

I don't think he meant LoLKing uses an exclusive API, but rather use the official API for everything possible (meaning they only use RTMP Calls for current games).

**Comment by Divi** [comment]
**Update:** RiotShmick said : 

---

## If your site is using RTMP, specator data, or other alternative sources than the API to power it, then yes, it will be in violation of the Terms of Service starting 10/1/14. The only exception is using spectator data to generate replay files. The data gathered from spectator data can not be used in other way.

So parsing the spectator files will be prohibed.

Source : https://developer.riotgames.com/discussion/riot-games-api/show/LQEpmLEX
